src/vm/precode.h

```diff
@@ -256,6 +260,12 @@ class Precode {
     MethodDesc *  GetMethodDesc(BOOL fSpeculative = FALSE);
     BOOL          IsCorrectMethodDesc(MethodDesc *  pMD);
 
+    // Determines if the precode is part of a native image
+    BOOL IsZapped();
+    // Determines if the precode is part of a native image
+    // Optimization if the caller already has the precode's MethodDesc available.
+    BOOL IsZapped(PTR_MethodDesc pMD);
```

Comment:
I do not think that this optimization is worth it. Fetching methoddesc from precode is cheap.